### ISSUE TITLE
Plugin breaks poetry 1.8.0

### ISSUE THREAD

**Issue by berthin** [mtkennerly/poetry-dynamic-versioning#170]
Hi,

Poetry released a new version (1.8.0) and when combined with poetry-dynamic-versioning, poetry breaks;

```
$ poetry --version
Poetry (version 1.8.0)

$ poetry self add "poetry-dynamic-versioning[plugin]"
Using version ^1.2.0 for poetry-dynamic-versioning
Updating dependencies
Resolving dependencies... (0.1s)
Warning: The locked version 1.5.0 for shellingham is a yanked version. Reason for being yanked: Incorrect package metadata

Package operations: 5 installs, 14 updates, 0 removals

  - Downgrading certifi (2024.2.2 -> 2022.9.24)
  - Downgrading charset-normalizer (3.3.2 -> 2.1.1)
  - Downgrading cryptography (42.0.5 -> 38.0.1)
  - Downgrading idna (3.6 -> 3.4)
  - Downgrading more-itertools (10.2.0 -> 9.0.0)
  - Installing pyparsing (3.0.9)
  - Downgrading urllib3 (2.2.1 -> 1.26.12)
  - Downgrading jaraco-classes (3.3.1 -> 3.2.3)
  - Downgrading msgpack (1.0.7 -> 1.0.4)
  - Downgrading packaging (23.2 -> 21.3)
  - Downgrading platformdirs (4.2.0 -> 3.11.0)
  - Downgrading requests (2.31.0 -> 2.28.1)
  - Installing markupsafe (2.1.1)
  - Downgrading pexpect (4.9.0 -> 4.8.0)
  - Downgrading shellingham (1.5.4 -> 1.5.0)
  - Downgrading tomlkit (0.12.3 -> 0.11.6)
  - Installing dunamai (1.19.2)
  - Installing jinja2 (3.1.2)
  - Installing poetry-dynamic-versioning (1.2.0)
Warning: The file chosen for install of shellingham 1.5.0 (shellingham-1.5.0-py2.py3-none-any.whl) is yanked. Reason for being yanked: Incorrect package metadata

$ poetry --version
No module named 'packaging.metadata'
```

**Comment by edgarrmondragon** [comment]
I think this was fixed in Poetry 1.8.1 by https://github.com/python-poetry/poetry/pull/9031

**Comment by mtkennerly** [comment]
Yeah, I've added a test for Poetry 1.8.1, and that version is working.

Thanks for reporting this :)
